src/commands/up.rs

```diff
@@ -187,10 +187,20 @@ impl UpCommand {
                 cmd.arg("--help-args-only");
             }
             TriggerExecOpts::Local { app, working_dir } => {
-                let locked_app = spin_trigger::locked::build_locked_app(app, &working_dir)?;
+                let locked_app = spin_trigger::locked::build_locked_app(app.clone(), &working_dir)?;
+
+                let state_dir = match app.info.origin {
+                    ApplicationOrigin::File(mut f) => {
+                        f.pop();
+                        f
```

Comment:
Maybe join `.spin` at this point?  That way when we have `--app-state-dir` we can just switch it in here, and won't need to worry about the trigger appending `.spin` to it regardless.

Reply:
done